lib/ansible/modules/cloud/amazon/aws_ses_identity_dkim.py

```diff
@@ -0,0 +1,223 @@
+        default: enabled
+        choices: [ 'enabled', 'disabled']
+        type: str
+requirements: [ 'botocore', 'boto3' ]
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# Note: to setup SES identities use the aws_ses_identity module.
+
+- name: Enable DKIM setting for identity example@example.com
+  aws_ses_identity_dkim:
+      identity: example@example.com
+
+- name: Enable DKIM setting for identity example.com
+  aws_ses_identity_dkim:
+    identity: example.com
+  register: dkim_results
+
+- name: Example how to aws_ses_identity_dkim results to set DNS records for validation (if you use Route53 service for DNS)
+  route53:
+    record: "{{ item }}._domainkey.example.com"
+    value: "{{ item }}.dkim.amazonses.com"
+    type: CNAME
+    zone: example.com
+  with_items: "{{ dkim_results.dkim_attributes.dkim_tokens }}"
+
+- name: Disable DKIM setting for identity example.com
+  aws_ses_identity_dkim:
+    identity: example.com
+    state: disabled
+'''
+
+RETURN = '''
+dkim_attributes:
+    description: Dictionary with DKIM information
+    type: complex
+    returned: always
+    sample: {
+        "dkim_tokens": [],
+        "dkim_enabled": "True",
+        "dkim_verification_status": "Pending"
+    }
+    contains:
+        dkim_tokens:
+            descritpion: Array of the 3 DKIM verification tokens needed for DNS records
+            type: list
+            sample: ['u5d5fa5twxzdni4aaoffdqvgnvku5kvu', '6xzw7n3abimypkyan5s6kuse4nymrfo2', 'nkwigx63eervm4hyja2lwzhjcfqyhka7']
+        dkim_enabled:
+            description: Boolean indicating if DKIM email sending is enabled or disabled for identity
+            type: bool
+            sample: True
+        dkim_verification_status:
+            description: Status of the DKIM verification process (DNS verification)
+            type: str
+            sample: 'Pending'
+'''
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict, AWSRetry
+
+import os
+import time
+
+try:
+    from botocore.exceptions import BotoCoreError, ClientError
+except ImportError:
+    pass  # caught by imported HAS_BOTO3
+
+
+def _get_state_check(state):
+    if state == 'enabled':
+        return lambda attributes: attributes['DkimEnabled'] and attributes['DkimTokens']
+    elif state == 'disabled':
+        return lambda attributes: not attributes['DkimEnabled']
+    elif state == 'any':
+        return lambda attributes: True
+    else:
+        raise RuntimeError('Unknown state ' + state)
+
+
+def get_identity_dkim_settings(module, client, identity, wait_for, retries=10, retry_delay=10):
+    for retry in range(1, retries + 1):
+        try:
+            if not module.check_mode:
+                response = client.get_identity_dkim_attributes(Identities=[identity])
+                dkim_attributes = response['DkimAttributes'][identity]
+            else:
+                dkim_attributes = {}
```

Comment:
Leaving `dkim_attributes` empty in check mode at the moment. Or you thinks it's better to set a dummy dict here? Like `dkim_attributes = { 'DkimEnabled': true }`? I think not as that would be potentiallz lying about the result...

Reply:
This function is being used both for getting the initial state as well as for getting the state of DKIM after a change.

I think the change her to not call get_identty_dkim in check mode is intended for the after the change call, but it's going to cause problems in the initial check about whether a change is needed.

As you've identified it's causing errors in the check, but also since the point of check mode is to check if a change would be made we really do need to call the real get_identity_dkim_attributes even if check mode is enabled.

So I'd suggest switching this back to caling get_identity_dkim attribute regardless of check_mode. The check_mode can be better handled in the main method, see comment below.